pkg/kubelet/server/stats/volume_stat_calculator.go

```diff
@@ -133,7 +134,14 @@ func (s *volumeStatCalculator) calcAndStoreStats() {
 	var ephemeralStats []stats.VolumeStats
 	var persistentStats []stats.VolumeStats
 	for name, v := range metricVolumes {
-		metric, err := v.GetMetrics()
+		metric, err := func() (*volume.Metrics, error) {
+			startTime := time.Now()
+			defer func() {
```

Comment:
nit: shouldn't this just be:
```golang
defer servermetrics.VolumeStatCalDuration.WithLabelValues(s.pod.Namespace, s.pod.Name, name).Set(servermetrics.SinceInSeconds(startTime))
```

Reply:
This will not work.

`defer` will execute the  `servermetrics.SinceInSeconds(startTime)` first and only defer the `.Set()` at the end.
So the `servermetrics.SinceInSeconds(startTime)` will always be a small duration like 0.001s.

Reply:
Updated.

As I wrapped it in a func.